### PR TITLE
Port pcd8544 bindings from .Net IoT

### PR DESCRIPTION
**Pull request by MrCSharp22**
## Description
- Port Pcd8544 binding from .Net IoT.
- Remove dependency on ImageSharp and add new graphic primitives (`Rectangle`, `Point`).
- Add a separate .Net 6 console app to convert images into pcd8544-compatible bitmap.

## Motivation and Context
I got my hands on a Pcd8544 LCD screen and well.. here we are.

- Closes nanoFramework/Home#860

## How Has This Been Tested?
- Manual testing by observing display outputs.
- Tested using an ESP-WROOM-32 Dev board and Pcd8544

## Types of changes
- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [X] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [X] My code follows the code style of this project (only if there are changes in source code).
- [X] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [X] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [X] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: devices/Pcd8544/README.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+## Documentation
+
+- Nokia5110 [datasheet](https://www.sparkfun.com/datasheets/LCD/Monochrome/Nokia5110.pdf)
+
+## Usage
+
+The screen is most of the time presented like this:
+
+![Nokia 5110 screen](./Nokia-5110-LCD-Pinout.png)
```
Comment (rkalwak) [inline devices/Pcd8544/README.md:13]: As this is nice that we normally do [Fritzing](https://fritzing.org/) schemas + export to png file.
You can find in the internet a free version of it or ping me on Discord ;)

File: devices/Pcd8544/version.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "$schema": "https://raw.githubusercontent.com/dotnet/Nerdbank.GitVersioning/master/src/NerdBank.GitVersioning/version.schema.json",
+  "version": "1.2",
```
Comment (rkalwak) [inline devices/Pcd8544/version.json:3]: That is new stuff right? Then version should be 1.0

File: src/IoT.Device.Graphics/DictionaryCharByte.cs [hunk 3]
```diff
@@ -97,5 +97,26 @@ public bool ContainsKey(char cr)
 
             return false;
         }
+
```
Comment (rkalwak) [inline src/IoT.Device.Graphics/DictionaryCharByte.cs:100]: Commented code should be removed if not needed or used ;)

File: devices/Pcd8544/samples/Pcd8544.samples/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("CSharp.BlankApplication")]
```
Comment (rkalwak) [inline devices/Pcd8544/samples/Pcd8544.samples/Properties/AssemblyInfo.cs:8]: Use minimal version of that file the same as for library, we don't need the rest of attributes

File: devices/Pcd8544/Pcd8544.cs [hunk 5]
```diff
@@ -0,0 +1,758 @@
...
+using System.Threading;
+using Iot.Device.Pcd8544Enums;
+using Iot.Device.CharacterLcd;
+using Iot.Device.Graphics;
+using System.Collections;
+
+namespace Iot.Device
+{
+    /// <summary>
+    /// PCD8544 - 48 × 84 pixels matrix LCD, famous Nokia 5110 screen
```
Comment (Ellerbach) [inline devices/Pcd8544/Pcd8544.cs:17]: I've seen you've added StyleCop but seems like it is not picking up some of the elements. In this one, the sentence should finish by a dot.

File: devices/Pcd8544/Pcd8544.cs [hunk 6]
```diff
@@ -0,0 +1,758 @@
...
+
+        /// <summary>
+        /// The bias for 0 to 7. Bias represent the voltage applied to the LCD. The highest, the darker the screen will be.
+        /// </summary>
+        public byte Bias
+        {
+            get => _bias;
+            set
+            {
+                _bias = value < 8 ? value : throw new ArgumentException($"Bias can't be more than 7");
```
Comment (Ellerbach) [inline devices/Pcd8544/Pcd8544.cs:215]: bonus point for all the exceptions: we are removing the text inside when there is nothing specific like in this case.
And we are using the /// intellisence comments instead. So would be really perfect if you can do it for those.

File: devices/Pcd8544/README.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+In this schema, the Chip Select (active low) is most of the time called CE. The SPI Clock has to be connected to CLK or SCLK. And the MOSI pin to DIN or MOSI. Names for the pins varies a bit.
+
+Reset pin is sometimes called RST.
+
+Supply voltage can vary, 3.3 or 5V, up to 7V is acceptable. Make sure that there is a current limiting resistor between the brightness led and the controller you'll plug. Most of the kits you buy have it. If you don't add one, you will damage the external brightness leds.
+
+The following example shows how to setup the screen with a PWM to control the brightness:
+
+```csharp
+            var resetPin = 32;
```
Comment (Ellerbach) [inline devices/Pcd8544/README.md:28]: you can move all to the left

File: devices/Pcd8544/README.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+Also you should `Refresh` the screen once you'll finish your drawing. All drawing are in the memory and needs to be pushed to the screen.
+
+### Displaying raw images
+
+The function SetByteMap allows you to draw anything, you'll just need to provide the raw buffer. Size is 504 bytes representing from the top left part columns of 8 pixels up to the right up to the next raw. A total of 6 raws are available with 84 columns each.
+
+Here is an example on how you can convert an existing image to extract the raw data:
+
+```csharp
+using Image<Rgba32> bitmapNokia = Image.Load<Rgba32>(Path.Combine("nokia_bw.bmp"));
```
Comment (Ellerbach) [inline devices/Pcd8544/README.md:98]: I guess you have to replace this part of the readme with your application Img2Pcd8544 code

**Review by Ellerbach — COMMENTED** [review]
Great start! Thanks

File: devices/Pcd8544/Pcd8544.cs (hunk 5 above)
Comment (MrCSharp22) [reply devices/Pcd8544/Pcd8544.cs:17]: Yeah, not sure what is going on here. It gave me some warnings the first time and now it is just quiet. I even re-installed the package. I'm just going to manually do that :)

File: src/IoT.Device.Graphics/DictionaryCharByte.cs (hunk 3 above)
Comment (MrCSharp22) [reply src/IoT.Device.Graphics/DictionaryCharByte.cs:100]: Forgot about this one. Thank you!

File: devices/Pcd8544/README.md (hunk 1 above)
Comment (MrCSharp22) [reply devices/Pcd8544/README.md:13]: Done

image

File: devices/Pcd8544/Pcd8544.cs (hunk 5 above)
Comment (MrCSharp22) [reply devices/Pcd8544/Pcd8544.cs]: Done!! Turns out StyleCop's C# parser was getting stuck on some of the new C# syntax that it doesn't support yet (maybe ever?) I removed that syntax and it started reporting the warnings again.

Comment (Ellerbach) [reply devices/Pcd8544/Pcd8544.cs]: Great! This version may not support that. It's the time we can move to SDK. Once there, we can use the "modern" version of StyleCop which is supporting the very new part of the language.

Comment (MrCSharp22) [reply devices/Pcd8544/Pcd8544.cs]: Is that something that is happening now across the project? Is there something for me to do for this PR?

File: devices/Pcd8544/README.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+bitmapLarge.Mutate(x => x.Resize(new Size(84, 48)));
+bitmapLarge.Mutate(x => x.BlackWhite());
+...
+```
+
+Note: you may want to reverse the colors first depending on what you want.
+
+This code is also available as a simple C# Console Application in Img2Pcd8544 and you can execute it using this command:
+
+```Img2Pcd8544 <PATH_TO_IMAGE_FILE>```
```
Comment (Ellerbach) [inline devices/Pcd8544/README.md:140]: only 1 apostrophe is needed or then make it on 3 lines

File: devices/Pcd8544/README.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+// Open an image and resize it
+using var image = Image.Load<Rgba32>(imageFilePath);
+bitmapLarge.Mutate(x => x.Resize(new Size(84, 48)));
+bitmapLarge.Mutate(x => x.BlackWhite());
+...
+```
+
+Note: you may want to reverse the colors first depending on what you want.
+
+This code is also available as a simple C# Console Application in Img2Pcd8544 and you can execute it using this command:
```
Comment (Ellerbach) [inline devices/Pcd8544/README.md:138]:
```suggestion
This code is also available as a simple C# Console Application in [Img2Pcd8544](./Img2Pcd8544) and you can execute it using this command:
```

**Review by Ellerbach — APPROVED** [review]
Very minor things in the readme. Then we should be good to merge!
Thanks again, great addition!

**Comment by MrCSharp22** [comment]
Thanks for your time reviewing this one @Ellerbach and @rkalwak 🤘

**Pull request merged**